File: backend/app/schemas/progress.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from uuid import UUID
from app.models.progress import ProgressStatus


class ProgressResponse(BaseModel):
    module_id: str
    current_lesson: int
    total_lessons: int
    status: ProgressStatus
    progress_percentage: int
    started_at: datetime
    updated_at: datetime
    completed_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True


class OverallProgressResponse(BaseModel):
    total_modules: int
    completed_modules: int
    in_progress_modules: int
    average_grade: Optional[float] = None
    modules: List[ProgressResponse]

File: frontend/src/api/progress.ts
import api from './axios'

export interface Progress {
  module_id: string
  current_lesson: number
  total_lessons: number
  status: string
  progress_percentage: number
  started_at: string
  updated_at: string
  completed_at?: string
}

export interface OverallProgress {
  total_modules: number
  completed_modules: number
  in_progress_modules: number
  average_grade?: number
  modules: Progress[]
}

export const getProgress = async (): Promise<OverallProgress> => {
  const response = await api.get<OverallProgress>('/progress')
  return response.data
}

export const getModuleProgress = async (moduleId: string): Promise<Progress> => {
  const response = await api.get<Progress>(`/progress/${moduleId}`)
  return response.data
}


